Include/internal/pycore_signal.h

```diff
@@ -38,12 +37,12 @@ PyAPI_FUNC(void) _Py_RestoreSignals(void);
 #define INVALID_FD (-1)
 
 struct _signals_runtime_state {
-    volatile struct {
-        _Py_atomic_int tripped;
+    struct {
+        int tripped;
         /* func is atomic to ensure that PyErr_SetInterrupt is async-signal-safe
          * (even though it would probably be otherwise, anyway).
          */
-        _Py_atomic_address func;
+        PyObject* func;
     } handlers[Py_NSIG];
 
     volatile struct {
```

Comment:
In a following PR, it may interesting to also change how 'wakeup' members are accessed. When I modified the code, I used "volatile" since I didn't want/couldn't use pycore_atomic API. Now with pyatomic.h, there are enough API to access members of this structure with atomic functions.

Reply:
+1